configure.ac

```diff
@@ -168,6 +169,62 @@ AS_IF([test "x$enable_exp_cq" == xyes],
 	])
 )
 
+#
+# VMAPOLL
+#
+AC_ARG_ENABLE(
+    [vmapoll],
+    AC_HELP_STRING(
+        [--enable-vmapoll],
+        [Enable alternative API to standard sockets API for receiving packets (default=no)]))
+AC_MSG_CHECKING(
+    [if vmapoll API is enabled])
+if test "x$enable_vmapoll" = xyes; then
+    AC_DEFINE([DEFINED_VMAPOLL], 1, [Define to 1 if vmapoll was enabled at configure time])
+    AC_MSG_RESULT([yes])
+else
+    AC_MSG_RESULT([no])
+fi
+
+#
+# Thread locking control
+#
+AC_ARG_ENABLE(
+    [thread-lock],
+    AC_HELP_STRING(
+        [--enable-thread-lock],
+        [Enable thread locking (default=yes)]))
+AC_MSG_CHECKING(
+    [if thread locking is enabled])
+if test "x$enable_thread_lock" = "xno"; then
+    if test "x$enable_vmapoll" = xyes; then
+        AC_DEFINE([DEFINED_NO_THREAD_LOCK], 1, [Define to 1 to disable thread locking])
+        AC_MSG_RESULT([no])
+    else
+        AC_MSG_RESULT([yes (vmapoll is not enabled)])
+    fi
+else
+    AC_MSG_RESULT([yes])
+fi
+
```

Comment:
The merge reflects the current functionality of experimental_256B branch
Improvements will be implemented after the merge.